include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -415,30 +510,57 @@ __pattern_search(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R1
 // search_n
 //------------------------------------------------------------------------
 
+template <typename _Tp>
+struct __pattern_search_n_fn
+{
+    _Tp __value;
+
+    template <typename _TValue1>
+    _Tp
+    operator()(_TValue1&&) const
+    {
+        return __value;
+    }
+};
+
 template <typename _BackendTag, typename _ExecutionPolicy, typename _Range, typename _Size, typename _Tp,
           typename _BinaryPredicate>
 oneapi::dpl::__internal::__difference_t<_Range>
 __pattern_search_n(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Range&& __rng, _Size __count,
                    const _Tp& __value, _BinaryPredicate __pred)
 {
+    __pattern_search_n_fn<_Tp> __fn{__value};
+
     //TODO: To consider definition a kind of special factory "multiple_view" (addition to standard "single_view").
     //The factory "multiple_view" would generate a range of N identical values.
     auto __s_rng = oneapi::dpl::experimental::ranges::views::iota(0, __count) |
-                   oneapi::dpl::experimental::ranges::views::transform([__value](auto) { return __value; });
+                   oneapi::dpl::experimental::ranges::views::transform(__fn);
 
     return __ranges::__pattern_search(__tag, std::forward<_ExecutionPolicy>(__exec), std::forward<_Range>(__rng), __s_rng,
                             __pred);
 }
 
 #if _ONEDPL_CPP20_RANGES_PRESENT
+template <typename _Pred, typename _Proj>
+struct __pattern_search_n_pred
```

Comment:
We already defined a type with the same functionality above. And we can re-use one.

Reply:
It's not correct: in this case we don't apply any `proj` operation to the second argument.